cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperatorTest.java

```diff
@@ -460,11 +459,13 @@ public void testNewWithMissingHardGoals(VertxTestContext context) throws IOExcep
                 kr).onComplete(context.succeeding(v -> context.verify(() -> {
                     // the resource moved from New to NotReady due to the error
                     KafkaRebalance kr1 = Crds.kafkaRebalanceOperation(kubernetesClient).inNamespace(CLUSTER_NAMESPACE).withName(RESOURCE_NAME).get();
-                    assertState(kr1, KafkaRebalanceState.NotReady, CruiseControlRestException.class,
+                    assertThat(kr1, State.hasState(KafkaRebalanceState.NotReady));
+                    Condition condition = kcrao.rebalanceStateCondition(kr1.getStatus());
+                    assertThat(Lists.newArrayList(condition), State.hasStateInConditions(KafkaRebalanceState.NotReady, CruiseControlRestException.class,
```

Comment:
You could make a more generic matcher hasStateInCondition
To handle this case.

hasStateInConditions would use this new method, i.e. matchers, calling matchers.

Reply:
But if I create a new static matcher, I will not be able to use `KafkaRebalanceAssemblyOperator kcrao` because it is not static. Am I missing something ?